src/Analysis/Ast/Impl/Analyzer/ModuleWalker.cs

```diff
@@ -242,57 +242,103 @@ public ModuleWalker(IServiceContainer services, IPythonModule module, PythonAst
 
                 var sourceVar = Eval.GlobalScope.Variables[v.Name];
                 var sourceType = sourceVar?.Value.GetPythonType();
-                
+
                 // If stub says 'Any' but we have better type, keep the current type.
                 if (!IsStubBetterType(sourceType, stubType)) {
-                    continue;;
+                    continue; ;
                 }
 
-                // If types are the classes, merge members.
-                // Otherwise, replace type from one from the stub.
-                if (sourceType is PythonClassType cls && Module.Equals(cls.DeclaringModule)) {
-                    // If class exists and belongs to this module, add or replace
-                    // its members with ones from the stub, preserving documentation.
-                    // Don't augment types that do not come from this module.
-                    foreach (var name in stubType.GetMemberNames()) {
-                        var stubMember = stubType.GetMember(name);
-                        var member = cls.GetMember(name);
-
-                        var memberType = member?.GetPythonType();
-                        var stubMemberType = stubMember.GetPythonType();
-
-                        if (builtins.Equals(memberType?.DeclaringModule) || builtins.Equals(stubMemberType.DeclaringModule)) {
-                            continue; // Leave builtins alone.
-                        }
-                        if (!IsStubBetterType(memberType, stubMemberType)) {
-                            continue;
+                // If type does not exist in module, but exists in stub, declare it unless it is an import.
+                // If types are the classes, merge members. Otherwise, replace type from one from the stub.
+                switch (sourceType) {
+                    case null:
+                        if (v.Source == VariableSource.Declaration) {
+                            Eval.DeclareVariable(v.Name, v.Value, v.Source);
```

Comment:
Bug was that we did not transfer variables from stub to the primary if primary did not have same variable declared. This caused problems restoring types since said variable could declare a class that the type uses in say, annotations.